projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -0,0 +1,154 @@
+#include "ads1015.h"
+// All the important data about ADS1015 is stored in ADS1015Storage storage by different functions.
+// Ads1015_init writes the desired settings to registers and inits the GPIO pin (ALRT/RDY).
+// From there, the user would start configuring channels with ads1015_configure_channel, and inside
+// this function an interrupt is registered on the ALRT/RDY pin with prv_interrupt_handler. Once the
+// interrupt is raised, prv_interrupt_handler reads the conversion values from the registers and
+// stores them in storage. The user could access this data by the provided read functions. These
+// functions get their data from storage and not directly by reading the registers. Also look at
+// ads1015_def for more information on macros and how registers are setup.
+#include <status.h>
+#include <stdio.h>
+#include "ads1015_def.h"
+#include "gpio_it.h"
+
+// reset the internal registers and enter a power-down state
+static StatusCode prv_reset(Ads1015Storage *storage) {
+  uint8_t reset = ADS1015_RESET_BYTE;
+  return i2c_write(storage->i2c_port, ADS1015_I2C_GENERAL_CALL, &reset, sizeof(reset));
+}
+
+// writes to register given upper and lower bytes
+static StatusCode prv_setup_register(Ads1015Storage *storage, uint8_t reg, uint8_t msb,
+                                     uint8_t lsb) {
+  uint8_t ads1015_setup_register[] = { reg, msb, lsb };
+  return i2c_write(storage->i2c_port, storage->i2c_addr + ADS1015_I2C_BASE_ADDRESS,
+                   ads1015_setup_register, SIZEOF_ARRAY(ads1015_setup_register));
+}
+
+// reads the register and stores the value in the given array
+static StatusCode prv_read_register(I2CPort i2c_port, Ads1015Address i2c_addr, uint8_t reg,
+                                    uint8_t *read, size_t size_read) {
```

Comment:
We usually tend to use `[x]`, `[x]_len` or similar just for consistency. It makes it easier to match parameters when you have more than one array.